tornado/httputil.py

```diff
@@ -479,6 +479,14 @@ class HTTPInputError(Exception):
     """
     pass
 
+class HTTPBadRequestInputError(HTTPInputError):
+    """Exception class for malformed HTTP requests from remote sources
+    that should result in a HTTP 400 response
```

Comment:
Now that we're making this change, I think most if not all uses of HTTPInputError should result in an HTTP 400 response. Did you notice any places where that was not the case? If not, I would just use HTTPInputError (and send the response `if not self.is_client`).